sheet_to_triples/rdf.py

```diff
@@ -18,11 +18,49 @@
 _GEO = (VM.atGeoPoint.toPython(), VM.atGeoPoly.toPython(), VM.name.toPython())
 
 
-def _cast_from_term(t):
+class Resolver:
+    #TODO: Make from_qname a method on this class. Will require rewriting
+    # calling code in other modules.
+    def __init__(self, graph):
+        self.graph = graph
+        namespaces = self.graph.namespace_manager.namespaces()
+        self.ns_match = '|'.join(
+            sorted(
+                [re.escape(str(ns)) for _, ns in namespaces],
+                key=len, reverse=True
+            )
+        )
+
+    def from_identifier(self, value):
+        if isinstance(value, rdflib.term.Identifier):
+            return value
+        prefix, _, tail = value.partition(':')
+        if tail:
+            namespace = self.graph.store.namespace(prefix)
+            if namespace:
+                # Rough hack to see if this is a sequence path, and create
+                if ' / ' in tail:
+                    return functools.reduce(
+                        operator.truediv, map(
+                            lambda v: from_qname(v, self.graph.store.namespace),
+                            value.split(' / ')))
+                return rdflib.URIRef(namespace + tail)
+        if re.match('http', value) and re.match(self.ns_match, value):
+            return rdflib.URIRef(value)
+
+        value = _norm(value)
+        # if ends with language tag, create a Literal with the appropriate lang
+        if re.search(r"@[a-z]{2}$", value):
+            inner = value[1:-4] if value[0] == '"' else value[:-3]
+            return rdflib.Literal(inner, lang=value[-2:])
+        return rdflib.Literal(value)
+
+
+def _cast_from_term(t, resolver):
```

Comment:
In this case I'd probably take the bound method as the arg rather than the instance.